pkg/crc/machine/generate_bundle_linux.go

```diff
@@ -0,0 +1,50 @@
+package machine
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/machine/bundle"
+	crcos "github.com/code-ready/crc/pkg/os"
+)
+
+func copyDiskImage(destDir string, crcBundleMetadata *bundle.CrcBundleInfo) error {
+	tmpDir := filepath.Join(constants.MachineCacheDir, "image-rebase")
+	_ = os.RemoveAll(tmpDir) // clean up before using it
+	if err := os.Mkdir(tmpDir, 0700); err != nil {
+		return err
+	}
+	defer func() {
+		_ = os.RemoveAll(tmpDir) // clean up after using it
+	}()
+
+	// Copy the disk image from the machine/crc folder
+	diskName := filepath.Base(crcBundleMetadata.GetDiskImagePath())
+	diskPath := filepath.Join(constants.MachineInstanceDir, constants.DefaultName, diskName)
+	if err := crcos.CopyFileContents(diskPath, filepath.Join(tmpDir, fmt.Sprintf("%s_back", diskName)), 0644); err != nil {
+		return err
+	}
+
+	// Copy the original image from cache folder
```

Comment:
I'd be interested in *why* we copy it, than something repeating what the code just below says :)
We copy the base image because this is where we are going to commit the changes which were made to the VM
We copy the overlay file because before doing the commit, we need to adjust the backing file.

I'm wondering if everything (copy/rebase/commit) could be done in one go with `qemu-img dd` or `qemu-img convert`, or maybe even some `virsh vol-*` command?

Reply:
> I'd be interested in _why_ we copy it, than something repeating what the code just below says :)
> We copy the base image because this is where we are going to commit the changes which were made to the VM
> We copy the overlay file because before doing the commit, we need to adjust the backing file.

Done.

> I'm wondering if everything (copy/rebase/commit) could be done in one go with `qemu-img dd` or `qemu-img convert`, or maybe even some `virsh vol-*` command?

This is same steps which we perform on snc side for adding the backing file to overlay one, not sure if it can be done with single command, need to look more around it.

Reply:
I ran some tests in the morning:

```
$ time virsh -c qemu:///system vol-clone --pool crc crc.qcow2 clone.qcow2
Vol clone.qcow2 cloned from crc.qcow2

real	0m40.244s
user	0m0.009s
sys	0m0.005s

# additional runs 'real' for qemu-img convert were about the same as virsh clone (~38s)
# even if data below is showing 49s for 'real'
```

```
$ time qemu-img convert -f qcow2 -O qcow2 ~/.crc/machines/crc/crc.qcow2 convert.qcow2

real	0m49.962s
user	0m0.525s
sys	0m9.613s
```

For comparison, copy+rebase+commit gives:
```
$ time sh -x  ./img.sh 
+ mkdir -p /tmp/test.qcow/
+ cp /home/teuf/.crc/machines/crc/crc.qcow2 /tmp/test.qcow/overlayImage.qcow2
+ cp /home/teuf/.crc/cache/crc_libvirt_4.7.2/crc.qcow2 /tmp/test.qcow/crc.qcow2
+ qemu-img rebase -F qcow2 -b /tmp/test.qcow/crc.qcow2 /tmp/test.qcow/overlayImage.qcow2
+ qemu-img commit /tmp/test.qcow/overlayImage.qcow2
Image committed.
+ cp /tmp/test.qcow/crc.qcow2 /home/teuf/snc.qcow2

real	2m29.638s
user	0m1.573s
sys	0m23.620s
```

And all 3 images are about the same size (I did not look more closely if they have the same content or not)
```
$ ls -al snc.qcow2 convert.qcow2 ~/.crc/machines/crc/clone.qcow2 
-rw-r--r--. 1 teuf teuf 17041063936 Apr  6 03:33 ~.crc/machines/crc/clone.qcow2
-rw-r--r--. 1 teuf teuf 17041063936 Apr  6 03:31 convert.qcow2
-rw-r--r--. 1 teuf teuf 17045651456 Apr  6 05:04 snc.qcow2
```

Reply:
But this code really belongs in a new machine driver API imo :)